### PR TITLE
fix(space): flexGap支持与不支持情况UI表现不一致问题

### PR DESCRIPTION
**Pull request by Estelle00**
不支持flexGap时宽度占用内容空间减少问题
当支持flexGap且出现split时表现与不支持是不一致问题

[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |           |
| 🇨🇳 Chinese |           |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by afc163** [comment]
请提供问题的描述和重现。

**Comment by Estelle00** [comment]
image

File: components/space/index.tsx [hunk 1]
```diff
@@ -123,12 +123,13 @@ const Space: React.FC<SpaceProps> = props => {
     // Patch for gap not support
     if (!supportFlexGap) {
       gapStyle.marginBottom = -verticalSize;
+      gapStyle[marginDirection] = -horizontalSize;
     }
   }
 
   if (supportFlexGap) {
     gapStyle.columnGap = horizontalSize;
```
Comment (sdhr27) [inline components/space/index.tsx:131]: 这里也要除以2吧

**Comment by sdhr27** [comment]
不过如果是一开始就漏掉这个逻辑的话，突然调小可能会导致很多用户的界面表现改变？会不会不大好。

**Comment by li-jia-nan** [comment]
Hey! I'm sorry we never got back to you about this but it seems this is now really outdated. If this is something that still makes sense to the latest master version of antd, please send a new PR with this change :)

**Pull request closed without merging**